### PR TITLE
removed HARDTOSHOOT from skeletal zombie line

### PR DESCRIPTION
**Pull request by OromisElf**
#### Summary
Bugfixes "removes the HARDTOSHOOT flag from the skeletal zombie line"

#### Purpose of change

The skeletal zombie, brute and juggernaut, which are covered in bone plates, has the hard to shoot flag, which doesn't make sense.

#### Describe the solution

removes that tag from those 3 zombie types

#### Describe alternatives you've considered

not being able to hit a wall of bone as if it was only bones instead of boneplates

#### Testing

gotta be honest with you: I didn't test it, I only linted it

#### Additional context

the inspiration for this came to me after missing a skeletal brute 12 times from 30-40 tiles away and WHEN I hit it was only grazing hits; which at first I though 'ah yes, a skeleton, my bullet goes right through' but then I remembered that this is not how bone plate covered zombies work

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/issues/23463#issuecomment-380943825

**Comment by OromisElf** [comment]
while that's a valid argument that I wasn't aware of, I feel like that's poorly reflected in their description, the skeletal shocker doesn't have hardtoshoot and with the current description of a skeletal zombie having his skin replaced by ossified tissue, the skeletal brute having grown bone plates over his skin and the skeletal juggernaut having continuously growing bones around its form it feels like a much higher chance to graze them instead of missing them feels more fitting (even though there's probably no flag for that)

**Comment by actual-nh** [comment]
The ones with bony **overgrowth** (basic skeletal zombie - note the "sickening cracking sounds" - and juggernaut) should not be smooth enough for streamlining vs bullets.

**Comment by souricelle** [comment]
The justification for streamlining doesn't make sense with most of the skeleton enemies and their armor already makes guns less of an option for them, a basic skeleton has 30(!) bullet armor and it only goes up fromt here.

**Comment by Hulkninja** [comment]
I thought this was a Heroes of Might and Magic reference, at first.

You know, the skeletons in that game tend to be hard to shoot because, well, they're all bones.

But if the skeletons are indeed COVERED in bones, that shouldn't really make them harder targets to hit, just armored targets.

At least that's how it seems like to me.

**Comment by actual-nh** [comment]
> The justification for streamlining doesn't make sense with most of the skeleton enemies and their armor already makes guns less of an option for them, a basic skeleton has 30(!) bullet armor and it only goes up from there.

There's something up here - the basic skeleton has 30, yes, but the skeletal brute has 24.

File: data/json/monsters/zed_skeletal.json [hunk 1]
```diff
@@ -71,7 +71,7 @@
     "upgrades": { "half_life": 12, "into": "mon_skeleton_hulk" },
     "death_drops": "default_zombie_clothes",
     "death_function": [ "NORMAL" ],
-    "flags": [ "SEES", "HEARS", "HARDTOSHOOT", "REVIVES", "NO_BREATHE", "POISON", "FILTHY" ]
+    "flags": [ "SEES", "HEARS", "REVIVES", "NO_BREATHE", "POISON", "FILTHY" ]
```
Comment (actual-nh) [inline data/json/monsters/zed_skeletal.json:74]:
```suggestion
    "flags": [ "SEES", "HEARS", "HARDTOSHOOT", "REVIVES", "NO_BREATHE", "POISON", "FILTHY" ]
```
On the grounds that its armor has a more reasonable potential of being smooth, from the description, and moreover it has a lower armor vs bullets than the skeletal zombie.

Comment (Termineitor244) [reply data/json/monsters/zed_skeletal.json:74]: I feel like the skeleton evolutionary line should:
A._ Be all focused around the concept of "Bones evolved in such a way that its hard to shoot"
B._ Be all focused around the concept of "Bone armor, remove hard to shoot"
Or maybe have 2 different evolutionary lines for the skeleton, each based around a different specialization idea.
Whatever the case, this kind of things should be mentioned in the description, its already very confusing as it is, and leaving only one of the skeletons with the "HARDTOSHOOT" flag can be pretty confusing for the player (Base skeleton is easy to hit, its evolution its not, and then the next evolution is easy to hit again).

Comment (actual-nh) [reply data/json/monsters/zed_skeletal.json:74]: Point.

Comment (OromisElf) [reply data/json/monsters/zed_skeletal.json:74]: so that the skeletal zombie has 6 more bullet armor than the skeletal zombie hulk?

Comment (actual-nh) [reply data/json/monsters/zed_skeletal.json:74]: Currently, the skeletal zombie (30 vs bullets) _**already**_ has 6 more bullet armor than the skeletal zombie hulk (24 vs bullets).

Comment (OromisElf) [reply data/json/monsters/zed_skeletal.json:74]: oh wait, you mean the brute, not the hulk, right? one sec

Comment (actual-nh) [reply data/json/monsters/zed_skeletal.json:74]: Whoops, sorry!

Comment (OromisElf) [reply data/json/monsters/zed_skeletal.json:74]: can I count this conversation as resolved? I agree with Termineitor244 that either the whole line should keep (and in the case of skeletal shocker get) the hardtoshoot flag or none of them should.

**Comment by OromisElf** [comment]
actually I'm gonna revert those changes and make a separate pr for that I think

**Comment by OromisElf** [comment]
new pr to change armor values: #48028

**Comment by kevingranade** [comment]
> while that's a valid argument that I wasn't aware of, I feel like that's poorly reflected in their description,

Feel free to PR  change to the description that makes it more clear, but I'm not interested in stripping this feature from the skeletal zombies.

**Pull request closed without merging**